app/models/order.rb

```diff
@@ -0,0 +1,50 @@
+class Order < ActiveRecord::Base
+  has_many :order_items
+  belongs_to :user
+
+  validates :order_state, presence: true
+  validates :name, presence: true, on: :update
+  validates :email, presence: true, on: :update
+  validates :address, presence: true, on: :update
+  validates :city, presence: true, on: :update
+  validates :state, presence: true, on: :update
+  validates :zip, presence: true, on: :update
+  validates :card_name, presence: true, on: :update
+  validates :last_four_cc, presence: true, on: :update
+  validates :cc_cvv, presence: true, on: :update
+  validates :billing_zip, presence: true, on: :update
+  validates :cc_exp_month, presence: true, on: :update
+  validates :cc_exp_year, presence: true, on: :update
```

Comment:
I love the `on: :update` clarification. That is a clean and specific touch!

One thing to note is that you can actually string these together if they all have the same requirements so for instance:

```validates :name, presence: true, on: :update
 validates :email, presence: true, on: :update
 validates :address, presence: true, on: :update```

Is the same as `validates :name, :email, :address, presence: true, on: :update`